com.unity.cinemachine/Runtime/Behaviours/CinemachineDecollider.cs

```diff
@@ -0,0 +1,344 @@
+#if CINEMACHINE_PHYSICS
+using System;
+using UnityEngine;
+
+namespace Unity.Cinemachine
+{
+    /// <summary>
+    /// An add-on module for CinemachineCamera that post-processes
+    /// the final position of the camera. Based on the supplied settings,
+    /// the Decollider will pull the camera out of any objects it is intersecting.
+    /// </summary>
+    [AddComponentMenu("Cinemachine/Procedural/Extensions/Cinemachine Decollider")]
+    [SaveDuringPlay]
+    [ExecuteAlways]
+    [DisallowMultipleComponent]
+    [HelpURL(Documentation.BaseURL + "manual/CinemachineDecollider.html")]
+    public class CinemachineDecollider : CinemachineExtension
+    {
+        /// <summary>
+        /// Camera will try to maintain this distance from any obstacle or terrain.
+        /// </summary>
+        [Tooltip("Camera will try to maintain this distance from any obstacle or terrain.  Increase it "
+            + "if necessary to keep the camera from clipping the near edge of obsacles.")]
+        public float CameraRadius = 0.4f;
+
+        /// <summary>Settings for pushing the camera out of intersecting objects</summary>
+        [Serializable]
+        public struct DecollisionSettings
+        {
+            /// <summary>When enabled, will attempt to push the camera out of intersecting objects</summary>
+            [Tooltip("When enabled, will attempt to push the camera out of intersecting objects")]
+            public bool Enabled;
+
+            /// <summary>Objects on these layers will be detected.</summary>
+            [Tooltip("Objects on these layers will be detected")]
+            public LayerMask ObstacleLayers;
+
+            /// <summary>
+            /// How gradually the camera returns to its normal position after having been corrected.
+            /// Higher numbers will move the camera more gradually back to normal.
+            /// </summary>
+            [Range(0, 10)]
+            [Tooltip("How gradually the camera returns to its normal position after having been corrected.  "
+                + "Higher numbers will move the camera more gradually back to normal.")]
+            public float Damping;
+
+            /// <summary>
+            /// Smoothing to apply to obstruction resolution.  Nearest camera point is held for at least this long.
+            /// </summary>
+            [Range(0, 2)]
+            [Tooltip("Smoothing to apply to obstruction resolution.  Nearest camera point is held for at least this long")]
+            public float SmoothingTime;
```

Comment:
Under which condition should damping and smoothing work?
![Mar-27-2024 14-17-32](https://github.com/Unity-Technologies/com.unity.cinemachine/assets/10277745/02e05067-504a-42b6-87cb-35e1a4a0b2cc)

Reply:
Do the tooltips help?

![image](https://github.com/Unity-Technologies/com.unity.cinemachine/assets/6424658/0e6c5369-5e1f-42f0-a9b0-16e6c8fa07fb)

It only damps when the camera returns to normal position, because in that case the path is clear.  If it damped when de-colliding then the camera would travel through the obstruction.

![image](https://github.com/Unity-Technologies/com.unity.cinemachine/assets/6424658/bd01ba58-4d42-4d31-aae7-b58dba7e6fb7)

Smoothing is exactly the same as in the deoccluder: the idea is that if the camera is sliding along a bumpy surface then it would hold the most pulled-in position for the smoothing time, and not fall back into the valleys.

Reply:
Ok yeah it make sense. Same would be nice to add this to the new doc.